utilities/DumpUnitary/Driver.cs

```diff
@@ -7,54 +7,77 @@
 
 namespace Quantum.DumpUnitary
 {
-    class Driver
+    public class Driver
     {
-        static void Main(string[] args)
-        {
-            double eps = 1E-5;          // the square of the absolute value of the amplitude has to be less than or equal to eps to be considered 0
+        const double eps = 1E-5;          // the square of the absolute value of the amplitude has to be less than or equal to eps to be considered 0
 
+        public static void runSim(
+            Func<QuantumSimulator, Int64, System.Threading.Tasks.Task> run,
+            ref string[] unitaryPattern,
+            ref string[] outputLines)
```

Comment:
The name `outputLines` is a bit vague - `unitaryPattern` is also the output, but the name seems to suggest that it's not. I'd use `matrixElements` instead, to stress what kind of output it is.